### PR TITLE
Feat/threads

### PR DESCRIPTION
**Pull request by oed**
This PR adds two main new features. Pinning of threads and a getThread REST endpoint. 

We start listening to a new message in the pinning-room, `SYNC_DB`. This message is for now only used for threads, but in the future it will replace the `OPEN_DB` message.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "express": "^4.16.4",
     "get-folder-size": "^2.0.0",
     "ipfs": "^0.33.1",
-    "orbit-db": "^0.19.9",
+    "orbit-db": "git://github.com/orbitdb/orbit-db.git#9d674766a54c7b949eca0d7e5ddc26b5486717cd",
```
Comment (zachferland) [inline package.json]: which orbit feature/fix is this for specifically? 

need to do the same for scaling, to get orbitdb cache options

File: src/pinning.js [hunk 2]
```diff
@@ -201,9 +223,13 @@ class Pinning {
...
   _onMessage (topic, data) {
     console.log(topic, data)
-    if (data.type === 'PIN_DB' && OrbitDB.isValidAddress(data.odbAddress)) {
-      this.openDB(data.odbAddress, this._openSubStoresAndSendHasResponse.bind(this), this._openSubStores.bind(this))
-      this.analytics.trackPinDB(data.odbAddress)
+    if (OrbitDB.isValidAddress(data.odbAddress)) {
+      if (data.type === 'PIN_DB') {
+        this.openDB(data.odbAddress, this._openSubStoresAndSendHasResponse.bind(this), this._openSubStores.bind(this))
+        this.analytics.trackPinDB(data.odbAddress)
+      } else if (data.type === 'SYNC_DB' && data.thread) {
```
Comment (zachferland) [inline src/pinning.js]: in future does sync db have to check and handle threads differently?

File: src/cacheService.js [hunk 3]
```diff
@@ -67,6 +70,23 @@ class CacheService {
...
+    const spaceName = req.query.space
+    const threadName = req.query.name
+
+    const fullName = namesTothreadName(spaceName, threadName)
+    const cachePosts = await this.cache.read(fullName)
+    let posts
+    try {
+      posts = cachePosts || await this.pinning.getThread(fullName)
+    } catch (e) {
+      res.status(500).send('Error: Failed to load posts')
```
Comment (zachferland) [inline src/cacheService.js]: can we differentiate between no thread/empty and code failure for error codes?

File: package.json (hunk 1 above)
Comment (oed) [reply package.json]: It's for the `determineAddress` function.

File: src/pinning.js (hunk 2 above)
Comment (oed) [reply src/pinning.js]: Yeah, I think soo. (see the updated pinning-room spec)

File: src/cacheService.js (hunk 3 above)
Comment (oed) [reply src/cacheService.js]: Hm, I'm not sure. If it's a thread that noone has posted to we will still open it and it will be empty.

Comment (zachferland) [reply src/cacheService.js]: ah right, so this 500 is for code failure? thats good then

File: package.json (hunk 1 above)
Comment (zachferland) [reply package.json]: ok cool, just checked this ref used the cache option too, so we have everything we need for new arch too

Comment (oed) [reply package.json]: Yup, I made sure as well :)

Comment (zachferland) [reply package.json]: also any concern using a different version of orbitdb in pinning than 3boxjs ? 19.9 was released last july, but doesn't look like to many major code changes up to this commit

Comment (oed) [reply package.json]: Nope, the PR for `determineAddress` should be the only change. 
However let's make sure to test this once it's deployed to develop :)

**Pull request merged**